nipype/interfaces/dtitk/registration.py

```diff
@@ -1,394 +1,419 @@
-
-                >>> import nipype.interfaces.dtitk as dtitk
-                >>> node = dtitk.DiffeoTask()
-                >>> node.inputs.fixed_file = 'diffusion.nii.gz'
-                >>> node.inputs.moving_file = 'diffusion2.nii.gz'
-                >>> node.inputs.mask = 'mask.nii.gz'
-                >>> node.inputs.legacy = 1
-                >>> node.inputs.n_iters = 6
-                >>> node.inputs.ftol = 0.002
-                >>> node.run() # doctest: +SKIP
-                """
     input_spec = DiffeoInputSpec
     output_spec = DiffeoOutputSpec
     _cmd = 'dti_diffeomorphic_reg'
 
     def _list_outputs(self):
         outputs = self.output_spec().get()
-        outputs['out_file_xfm'] = self.inputs.moving_file.replace(
-            '.nii.gz', '_diffeo.df.nii.gz')
-        outputs['out_file'] = self.inputs.moving_file.replace(
-            '.nii.gz', '_diffeo.nii.gz')
+        splitlist = os.path.basename(self.inputs.moving_file).split('.')
+        basename = splitlist[0]
+        termination = '.'+'.'.join(splitlist[1:])
+        outputs['out_file_xfm'] = basename+'_diffeo.df'+termination
+        outputs['out_file'] = basename+'_diffeo'+termination
         return outputs
 
 
 class ComposeXfmInputSpec(CommandLineInputSpec):
-    in_df = traits.Str(desc='diffeomorphic file.df.nii.gz', exists=True,
-                       mandatory=False, position=1, argstr="-df %s")
-    in_aff = traits.Str(desc='affine file.aff', exists=True, mandatory=False,
-                        position=0, argstr="-aff %s")
-    out_file = traits.Str(desc='output_path', exists=True, mandatory=False,
-                          position=2, argstr="-out %s",  name_source="in_df",
-                          name_template="%s_comboaff.nii.gz")
+    in_df = File(desc='diffeomorphic warp file', exists=True,
+                 argstr="-df %s", copyfile=False, mandatory=True)
+    in_aff = File(desc='affine_xfm.aff', exists=True,
+                  argstr="-aff %s", mandatory=True)
+    out_file = traits.Str(desc='output_path', exists=True,
+                          argstr="-out %s",  name_source="in_df",
+                          name_template="%s_aff.df", keep_extension=True)
 
 
 class ComposeXfmOutputSpec(TraitedSpec):
-    out_file = traits.File(desc='cheese', exists=True)
+    out_file = File(exists=True)
 
 
 class ComposeXfmTask(CommandLineDtitk):
     """
      Combines diffeomorphic and affine transforms
 
-                Example
-                -------
+    Example
+    -------
 
-                >>> import nipype.interfaces.dtitk as dtitk
-                >>> node = dtitk.ComposeXfmTask()
-                >>> node.inputs.in_df = 'ants_Warp.nii.gz'
-                >>> node.inputs.in_aff= 'ants_Affine.txt'
-                >>> node.run() # doctest: +SKIP
-                """
+    >>> import nipype.interfaces.dtitk as dtitk
+    >>> node = dtitk.ComposeXfmTask()
+    >>> node.inputs.in_df = 'im1.nii'
+    >>> node.inputs.in_aff= 'im2.nii'
```

Comment:
It would probably be more informative to use a file of the right type (e.g. `*.aff`). Note that you can use any filename in [`nipype/testing/data/`](https://github.com/nipy/nipype/tree/master/nipype/testing/data) (and you can add empty files to that directory to make these usefully realistic).